pkg/controller/backup_deletion_controller.go

```diff
@@ -312,6 +319,27 @@ func (c *backupDeletionController) processRequest(req *v1.DeleteBackupRequest) e
 	return nil
 }
 
+func (c *backupDeletionController) deleteBackupFromStorage(backup *v1.Backup, log *logrus.Entry) error {
+	pluginManager := c.newPluginManager(log, log.Level, c.pluginRegistry)
+	defer pluginManager.CleanupClients()
+
+	backupLocation, err := c.backupLocationLister.BackupStorageLocations(backup.Namespace).Get(backup.Spec.StorageLocation)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	objectStore, err := getObjectStoreForLocation(backupLocation, pluginManager)
+	if err != nil {
+		return err
```

Comment:
The pattern we follow is that when Ark code is creating an error, we use `github.com/pkg/errors` to create it (which automatically includes a stack), and if we're handling an error returned from external code, we wrap it. In this case, since we're handling an error from an Ark function, it should already be wrapped/with a stack so we don't need to do it again.